### PR TITLE
Fix host being always initially created in main folder.

### PR DESCRIPTION
**Pull request by lgetwan**
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Hosts are always created in main folder if previous state was `absent`.

## What is the new behavior?

- Hosts are created in the folder that is provided in the parameters

### CODE REVIEW

File: changelogs/fragments/fix_host_initial_folder.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+ - Rule module - Fix empty rule conditions.
```
Comment (robin-checkmk) [inline changelogs/fragments/fix_host_initial_folder.yml:2]: This looks like the wrong changelog fragment to me. :)

**Comment by robin-checkmk** [comment]
Fix confirmed in Checkmk 2.2.0p25. Now testing 2.3.0b6.

**Comment by msekania** [comment]
@lgetwan, 

thanks for quick bug fix!

may be it would better also to include test of this behavior in test suite. So that the future changes do not miss this point.

**Comment by robin-checkmk** [comment]
> may be it would better also to include test of this behavior in test suite. So that the future changes do not miss this point.

I like that, but have no time to implement it. If you are up to the task, please go ahead.
I will merge this PR and release a bug fix release soonish though.

**Review by robin-checkmk — APPROVED** [review]

**Comment by msekania** [comment]
@robin-checkmk,

I have quickly extended the tests, see #599.

**Comment by robin-checkmk** [comment]
Apparently the new tests in #599 do not work with Checkmk 2.0.0. I have not started digging yet, but as we will be dropping support for 2.0.0 from the collection soon, I would not invest too much time. @msekania if you can quickly see what is wrong, that would be great. Otherwise, I might have to remove testing for 2.0.0 here, so we can get the bug fix release on the road.

So by the end of tomorrow, we either have a solution, or will remove 2.0.0 from the testing scope.

**Comment by msekania** [comment]
@robin-checkmk, 

Yes I see the errors.

Unfortunately, I do not have a quick fix to the tests, except skipping it for version 2.0.0.

I am not quite sure were the problem is. From the automatic test output one can see that the hosts are created in the correct folders. I am not sure though, that the host lookup plugin extracts folder field correctly for version 2.0.0.

**Comment by robin-checkmk** [comment]
I disabled the 2.0.0 tests for the `host` module for now.

**Pull request merged**